app/ui/setup.py

```diff
@@ -172,3 +184,208 @@ def sndk_url(account: str, user: str, region: str) -> str:
+            print("Successfully created API integration.")
+            st.code(
+                """
+                "Successfully created API integration. Please run the following command in your Snowflake account"
+                """
+            )
+        else:
+            print("Please run the following command in your Snowflake account:")
+            setup_func = "ADMIN.SETUP_REGISTER_TENANT_FUNC()"
+            gateway_prefixes = f"'{ef_url}'"
+            st.code(
+                generate_code_to_setup_external_func(
+                    app_name, gateway_prefixes, api_integration_name, setup_func
+                )
+            )
+
+    st.markdown(
+        """
+        2. To create a free Sundeck account, Please run the following code in your Snowflake account.
+
+        """
+    )
+    st.code(
+        generate_code_to_create_sundeck_account(
+            db=app_name, sf_region=sf_region, sd_deployment=sd_deployment
+        )
+    )
+    st.button("Refresh Status", on_click=config.refresh, key="refresh2")
+
+
+def generate_code_to_create_sundeck_account(
+    db: str, sf_region: str, sd_deployment: str, name: str = "SUNDECK_OAUTH2"
+) -> str:
+    return f"""
+BEGIN  -- Create security integration and Sundeck account
+{generate_security_integration_code(sf_region, sd_deployment, name)}
+{generate_register_tenant_code(db, name)}
+END
+"""
+
+
+def generate_security_integration_code(
+    sf_region: str, sd_deployment: str, name: str
+) -> str:
+    redirect_url = get_redirect_url_for_security_integration(sf_region, sd_deployment)
+    return (
+        f"create security integration if not exists {name} type=oauth enabled=true oauth_client=CUSTOM "
+        + f"oauth_client_type='CONFIDENTIAL' oauth_redirect_uri='{redirect_url}' oauth_issue_refresh_tokens=true "
+        + "oauth_refresh_token_validity=86400 pre_authorized_roles_list = ('PUBLIC'); "
+    )
+
+
+def generate_register_tenant_code(db: str, security_integration_name: str) -> str:
+    return f"""
+let oauth_info variant := (parse_json(SYSTEM$SHOW_OAUTH_CLIENT_SECRETS('{security_integration_name}')));
+let tenantInfo object := {db}.tools.register_tenant(:oauth_info:OAUTH_CLIENT_ID, :oauth_info:OAUTH_CLIENT_SECRET);
+CALL {db}.admin.setup_sundeck_tenant_url(:tenantInfo:sundeckTenantUrl, :tenantInfo:sundeckUdfToken);
+let rs resultset := (select 'Go to Sundeck UI' As msg, :tenantInfo:sundeckTenantUrl::string as url);
+return table(rs);"""
+
+
+def generate_code_to_setup_external_func(
+    app_name: str, gateway_prefixes: str, api_integration_name: str, setup_func: str
+) -> str:
+    return (
+        f"""
+BEGIN
+    CREATE OR REPLACE API INTEGRATION {api_integration_name} api_provider = aws_api_gateway """
+        + f""" api_aws_role_arn = '{OPSCENTER_ROLE_ARN}' api_allowed_prefixes = ({gateway_prefixes}) enabled = true;
+    GRANT USAGE ON INTEGRATION {api_integration_name} TO APPLICATION "{app_name}";
+    CALL {setup_func};
+END;
+"""
+    )
```

Comment:
nit unnecessary parens?
```suggestion
    return
        f"""
BEGIN
    CREATE OR REPLACE API INTEGRATION {api_integration_name} api_provider = aws_api_gateway """
        + f""" api_aws_role_arn = '{OPSCENTER_ROLE_ARN}' api_allowed_prefixes = ({gateway_prefixes}) enabled = true;
    GRANT USAGE ON INTEGRATION {api_integration_name} TO APPLICATION "{app_name}";
    CALL {setup_func};
END;
"""
```

Reply:
precommit (black) is not happy about it. It is adding parens